runtime/gc_base/StringTable.cpp

```diff
@@ -666,20 +718,28 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 				}
 			}
 		} else {
-			if (compressStrings && isASCII) {
+			if (compressStrings && isASCIIorLatin1) {
 				UDATA lastSlash = 0;
 				if (translateSlashes) {
-					for (UDATA i = 0; i < length; ++i) {
-						U_8 c = data[i];
-						if ('/' == c) {
-							lastSlash = i;
-							c = '.';
+					if (isASCII) {
+						for (UDATA i = 0; i < length; ++i) {
+							U_8 c = data[i];
+							if ('/' == c) {
+								lastSlash = i;
+								c = '.';
+							}
+							J9JAVAARRAYOFBYTE_STORE(vmThread, charArray, i, c);
 						}
-						J9JAVAARRAYOFBYTE_STORE(vmThread, charArray, i, c);
+					} else {
+						lastSlash = storeLatin1ByteArrayhelper(vmThread, data, length, charArray, true);
 					}
 				} else {
-					for (UDATA i = 0; i < length; ++i) {
-						J9JAVAARRAYOFBYTE_STORE(vmThread, charArray, i, data[i]);
+					if (isASCII) {
+						for (UDATA i = 0; i < length; ++i) {
+							J9JAVAARRAYOFBYTE_STORE(vmThread, charArray, i, data[i]);
+						}
+					} else {
+						lastSlash = storeLatin1ByteArrayhelper(vmThread, data, length, charArray, false);
 					}
 				}
```

Comment:
Please help me understand why `charArray` actually refers to a byte array in this block; it doesn't appear to be guarded by `J9_ARE_ANY_BITS_SET(vm->runtimeFlags, J9_RUNTIME_STRING_BYTE_ARRAY)`.

Reply:
This is a bit of a hack - the primitive array macros work on any primitive array, so if we're storing bytes (either in a byte or char array), the byte macros are used.

Reply:
I agree with the 'hack' label, but thanks for the explanation.